# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/streamconv/nsIStreamConverter.idl
# Commit: 3b150d676b31
# Full Hash: 3b150d676b3122bffa703813876d1fbc35e10d17
# Author: Randell Jesup <rjesup@mozilla.com>
# Date: 2024-03-19 16:41:28
# Regressor Bug: 1356686
# File Overlap Count: 1
# Description:
#   Bug 1356686 - Do decompression off main thread in content r=necko-reviewers,valentin,extension-reviewers,robwu
#   
#   Content processes will now always retarget delivery of OnDataAvailable for Http
#   channels off the main thread. Consumers that were previously redirecting
#   off-main thread are not affected and their retargeting will stick, but any
# ==============================================================================

diff -r 54754a50d992 -r 3b150d676b31 netwerk/streamconv/nsIStreamConverter.idl
--- a/netwerk/streamconv/nsIStreamConverter.idl	Mon Mar 18 20:04:00 2024 +0000
+++ b/netwerk/streamconv/nsIStreamConverter.idl	Mon Mar 18 20:04:01 2024 +0000
@@ -96,6 +96,12 @@
                           in nsISupports aCtxt);
 
     /**
+     * This is called after the request has installed the stream converter as its listener
+     * giving the stream converter the option to request retargetting of onDataAvailable.
+     */
+    void maybeRetarget(in nsIRequest request);
+
+    /**
      * Returns the content type that the stream listener passed to asyncConvertData will
      * see on the channel if the conversion is being done from aFromType to * /*.
      *